### PR TITLE
Fix install when not building hipblas-bench

### PR DESCRIPTION
**Pull request by cgmb**
The hipblas-common target is only used by hipblas-bench and will therefore not be built during a build of only the library and tests. In that case, the install will fail.

This fixes the error by not installing hipblas-common unless hipblas-bench is to be built.

### CODE REVIEW

**Review by daineAMD — APPROVED** [review]
This won't be tested by CI, so approving assuming you've done the appropriate testing locally, I can try it out as well but LGTM.

File: clients/CMakeLists.txt
Comment (TorreZuk) [inline clients/CMakeLists.txt:139]: This looks wrong, if guarding it should be if( BUILD_CLIENTS_BENCHMARKS OR BUILD_CLIENTS_TESTS) etc.

**Review by TorreZuk — CHANGES_REQUESTED** [review]
Guard needs to be fixed

File: clients/CMakeLists.txt
Comment (cgmb) [reply clients/CMakeLists.txt:139]: hipblas-test is not marked as depending on clients-common, so that would cause the build to fail.

Comment (TorreZuk) [reply clients/CMakeLists.txt:139]: Probably can fix it all up after samples is merged

**Comment by TorreZuk** [comment]
Added deps on common and this change in linked PR so closing this one.

**Pull request closed without merging**
